lambda/shared/src/main/scala/feral/lambda/events/APIGatewayV2WebSocketEvent.scala

```diff
@@ -0,0 +1,304 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.events
+
+import io.circe.Decoder
+
+
+sealed abstract class APIGatewayV2WebSocketEvent {
+  def resource: String
+  def path: String
+  def httpMethod: String
+  def headers: Map[String, String]
+  def multiValueHeaders: Map[String, List[String]]
+  def queryStringParameters: Map[String, String]
+  def multiValueQueryStringParameters: Map[String, List[String]]
+  def pathParameters: Map[String, String]
+  def stageVariables: Map[String, String]
+  def requestContext: WebSocketRequestContext
+  def body: String
+  def isBase64Encoded: Boolean 
+}
+
+object APIGatewayV2WebSocketEvent {
+  def apply(
+    resource: String,
+    path: String,
+    httpMethod: String,
+    headers: Map[String, String],
+    multiValueHeaders: Map[String, List[String]],
+    queryStringParameters: Map[String, String],
+    multiValueQueryStringParameters: Map[String, List[String]],
+    pathParameters: Map[String, String],
+    stageVariables: Map[String, String],
+    requestContext: WebSocketRequestContext,
+    body: String,
+    isBase64Encoded: Boolean
+  ): APIGatewayV2WebSocketEvent =
+    new Impl(
+      resource,
+      path,
+      httpMethod,
+      headers,
+      multiValueHeaders,
+      queryStringParameters,
+      multiValueQueryStringParameters,
+      pathParameters,
+      stageVariables,
+      requestContext,
+      body,
+      isBase64Encoded
+    )
+  
+  implicit val decoder: Decoder[APIGatewayV2WebSocketEvent] = Decoder.forProduct12(
+    "resource",
+    "path",
+    "httpMethod",
+    "headers",
+    "multiValueHeaders",
+    "queryStringParameters",
+    "multiValueQueryStringParameters",
+    "pathParameters",
+    "stageVariables",
+    "requestContext",
+    "body",
+    "isBase64Encoded"
+  )(APIGatewayV2WebSocketEvent.apply)
+
+  private final case class Impl(
+    resource: String,
+    path: String,
+    httpMethod: String,
+    headers: Map[String, String],
+    multiValueHeaders: Map[String, List[String]],
+    queryStringParameters: Map[String, String],
+    multiValueQueryStringParameters: Map[String, List[String]],
+    pathParameters: Map[String, String],
+    stageVariables: Map[String, String],
+    requestContext: WebSocketRequestContext,
+    body: String,
+    isBase64Encoded: Boolean
+  ) extends APIGatewayV2WebSocketEvent {
+    override def productPrefix = "APIGatewayV2WebSocketEvent"
+  }
+}
+
+sealed abstract class WebSocketRequestIdentity {

```

Comment:
I didn't look too carefully. How similar is this one to the existing `RequestIdentity`? If they are the same maybe we can share the classes instead of implementing it twice.